### PR TITLE
feat: Updating Windows VHD build files to support building for multiple OS versions

### PR DESCRIPTION
**Pull request by marosset**
**Reason for Change**:

**Issue Fixed**:

**Credit Where Due**:

Does this change contain code from or inspired by another project?

- [x] No
- [ ] Yes

If "Yes," did you notify that project's maintainers and provide attribution?

- [ ] No
- [ ] Yes

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

**Comment by marosset** [comment]
I decided to not convert the Windows vhd packer file to use the HashiCorp Config Langauge becasue this is a smaller set of changes that accomplishes the same goals.

File: vhd/packer/windows-vhd-builder.json [hunk 1]
```diff
@@ -23,12 +28,12 @@
...
-            "image_publisher": "MicrosoftWindowsServer",
-            "image_offer": "WindowsServer",
-            "image_sku": "2019-Datacenter-Core-smalldisk",
-            "image_version": "17763.1339.2007101755",
+            "image_publisher": "{{user `image_publisher`}}",
+            "image_offer": "{{user `image_offer`}}",
+            "image_sku": "{{user `image_sku`}}",
+            "image_version": "{{user `image_version`}}",
             "resource_group_name": "{{user `resource_group_name`}}",
-            "capture_container_name": "aksengine-vhds-windows-ws2019",
```
Comment (CecileRobertMichon) [inline vhd/packer/windows-vhd-builder.json:31]: this will change from `ws2019` to `2019` by default, is that okay?

Comment (marosset) [reply vhd/packer/windows-vhd-builder.json:31]: I think it should be ok. This is only used in the paths in the storage account the VHD gets captured into correct?

**Comment by marosset** [comment]
Let me revert chagnes tot he scale.go file.. not sure how those got picked up...

**Comment by CecileRobertMichon** [comment]
it's a github beta feature `Unchanged files with check annotations`, no changes were pushed, it's just showing them to reviewers

**Comment by marosset** [comment]
First time I saw that. Thanks.

File: vhd/packer/windows-vhd-builder.json (hunk 1 above)
Comment (CecileRobertMichon) [reply vhd/packer/windows-vhd-builder.json:31]: yeah should be fine, we get the URI directly from the output so we don't expect it to be in a specific place 👍

**Review by CecileRobertMichon — APPROVED** [review]
the packer/pipeline parts lgtm

File: .pipelines/vhd-builder-windows.yaml [hunk 2]
```diff
@@ -40,6 +40,7 @@ jobs:
       -e BUILD_ID=$(Build.BuildId) \
       -e BUILD_NUMBER=$(Build.BuildNumber) \
       -e CONTAINER_RUNTIME=$(CONTAINER_RUNTIME) \
+      -e WINDOWS-SERVER-VERSION=2019 \
```
Comment (jsturtevant) [inline .pipelines/vhd-builder-windows.yaml:43]: should 2019 be an environment variable so it can be configured on the job?

Comment (marosset) [reply .pipelines/vhd-builder-windows.yaml:43]: Possibly. We still need to have a discussion around what images aks-engine should actually be producing. If we want to produce more than just Windows Server 2019 + Docker I think I would rather just update the pipepline to build all the images as different stages and avoid the need to trigger multiple pipelines.
I'm hoping we can resolve that with a different PR tho.

**Pull request merged**